### DISCUSSION TITLE
Has HP10 in osu!mania always been supposed to make you play with a "battery life"?

### DISCUSSION THREAD

**Discussion by Clpsplug** [ppy/osu#25645] [Q&A] [answered]
I could not tell if this was a bug or not, so let me open a discussion here.

Since the last **Lazer** update (2023.1130), I suddenly started failing a map that I could previously pass. I checked what was happening, and there seemed to be a change that matched the health drain behaviour in osu! and osu!catch to that of stable (ppy/osu#25418 and ppy/osu#25563).  
I'm unsure how that affected osu!mania, but I feel the HP has become significantly harsher than the last update. Here are some observations:

* HP 9, which most of the "Insane" and "Extra"-level maps have, feels like how a HARD gauge in most rhythm games works. You lose a lot on MISS, and you must combo a TON to gain the slightest of your HP back (even PERFECTs' contribution looks minuscule), but full recovery is always possible.
* HP 10 is the hardest in terms of how HP can ever behave. The HP bar essentially becomes a "battery bar", in which you still lose a lot, but you **never gain the lost health back**. In other words, if you score a set amount of MEH (50) or MISS, you're instantly dead. I noticed this behaviour by applying HR to the HP 7.5 map.

Here's me playing a map on HR. The mod made this map have HP10. My error bar placement blocks most of the HP bar from the view, but I kept my HP barely enough for a while so that I could demonstrate the lack of recovery in HP10.
Click the GIF for the full FPS footage:  
[![Image from Gyazo](https://i.gyazo.com/55ecbbcf8808a91bca3fe58a0b41baf8.gif)](https://gyazo.com/55ecbbcf8808a91bca3fe58a0b41baf8)

The same behaviours are observed when I use Difficulty Adjust to set the map HP to 9-10.

I haven't been following the recent game balance changes too much (started playing again about two months ago), plus I don't have an environment I can play stable on, so I don't know how the past HP bars have worked, but HP 10 forcing you into a battery gauge sounds a bit too harsh. After all, it is the hardest setting possible, so a very harsh HP penalty might be understandable, but the complete lack of recovery in HP10 looked suspicious. Plus, I don't think this was the case before the update when I tried HR on maps that I wanted to improve accuracy on.

Has HP 10 been meant to behave in that way? If it's not, I'll go ahead and create an issue.

**Comment by Clpsplug** [comment]
Sorry, I must turn this into an issue without the dev's reply. I dug into it a bit further, and I realized that if the HP of the map is 10, **even osu!topus (AT), who can always get PERFECTs, loses health ever so slightly over time and potentially fails mid-way if the song is long enough!** I genuinely do not believe this is a correct behaviour now.

**Comment by bdach** [comment]
@smoogipoo probably something you might want to look at.

**Comment by Clpsplug** [comment]
Hey folks, I've turned this into an issue. With some more tests, I now believe that the main issue here is that the passive health drain is back somehow. Harsher health recovery might also be something we're interested in, so I noted that as well there: https://github.com/ppy/osu/issues/25648.

**Comment by smoogipoo** [comment, marked as answer]
Will be addressed by https://github.com/ppy/osu/pull/25649
